passes/cmds/rename.cc

```diff
@@ -210,114 +210,98 @@ struct RenamePass : public Pass {
 		{
 			extra_args(args, argidx, design);
 
-			for (auto &mod : design->modules_)
+			for (auto module : design->selected_modules())
 			{
 				int counter = 0;
+				std::map<RTLIL::Wire *, IdString> new_wire_names;
+				std::map<RTLIL::Cell *, IdString> new_cell_names;
```

Comment:
Prefer dict to map

Reply:
Besides:
https://github.com/YosysHQ/yosys/blob/840bb170896c1acbbd321c4214df1f3b15108466/CodingReadme#L53-L80
using a `std::map<T*,U>` and then iterating over will lead to nondeterminism.